frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -3374,6 +3421,12 @@ gatherAndFilterCandidates(Context* context,
                           ForwardingInfoVec& forwardingInfo,
                           std::vector<ApplicabilityResult>* rejected) {
   CandidatesVec candidates;
+  // Last resort candidates, grouped by without-POI, then from innermost POI
+  // scope outward, then forwading.
+  std::vector<CandidatesVec> lrcGroups;
+  // Analogous last resort candidate handling for forwarding, if needed.
+  std::vector<CandidatesVec> forwardingLrcGroups;
+  std::vector<ForwardingInfoVec> forwardingLrcForwardingTo;
```

Comment:
Please _consider_ using some kind of data structure to contain this info (non-POI, POI*, forwarding*) instead of a flat list and relying on indexing. I don't know if it will improve the code, but this (and the subsequent special-casing of `i == 0` later on) is a bit hard to read. Feel free to resolve if you have considered it and it does not help :)

Reply:
I did consider it but was a little torn since I couldn't come up with a solution that wouldn't require replacing the `i == 0` special case in the loop with either another special case, or some nearly-duplicated code outside the loop. However I do think the downside of a flat list relying on an arbitrary order being followed is worse. I've implemented a struct to organize the LRC groups now.